### ISSUE TITLE
select2 Error Handling

### ISSUE THREAD

**Issue by CWSpear** [AngularAgility/AngularAgility#99]
Am I missing something, or if I use the select2 module with a field group, does it not handle errors the same way?

i.e., take this code: 

``` js
<div aa-select2="static.select2StatesConfig" aa-field-group="address.state" required></div>
```

If I focus the field and then move out, the error message doesn't appear like it does for regular input fields.

It would be nice if it 1) showed errors on blur like other inputs and 2) it got the `aa-had-focus` (or whatever that class is) put on the `.select2-container` element so that we can style it.

This is out of scope of the original issue, but while we're in there, since select2 is a faux element, it would be awesome if it could add something like an `aa-has-focus` class (note **has** instead of **had**) while the element has focus so that we can style it to match `:focus` styles for other inputs.

**Comment by johnculviner** [comment]
Yeah select2 is pretty jenky if you try to mix and match the two since its really just a big, crazy jQuery control. I'm not using it much of late but if you find a fix feel free to send a PR and I'll merge it in. Thanks!

**Comment by CWSpear** [comment]
Yeah, we ended up moving away from select2 because it's not very mobile friendly. Without the urgency to use it on a project, I may never get around to this!

**Comment by smajl** [comment]
Two parts of the same lib don't work together? That's a huge let down.. :(

@johnculviner So what do you use for some advanced select functions in Angular environment in combination with aa-form-extensions (so it works together)? You stick to simple usage with native select element or https://github.com/angular-ui/ui-select maybe?

**Comment by johnculviner** [comment]
What were you having issue with? I have some code like this that was working:

```
<div aa-select2="select2config" aa-field-group="my.fieldName" required></div>
```

**Comment by johnculviner** [comment]
Never mind. Some of the finer things as mentioned at the top of the issue. Yeah it works pretty well but its not perfect

**Comment by johnculviner** [comment]
@smajl you can always do the validation stuff manually if something isn't working for you. not ideal but angular agility was really ment to help with the 80-90% and not work perfectly all the time for all advanced scenarios. Works great with normal  HTML elements though
